x/tokenomics/token_logic_module/validator_distribution_test.go

```diff
@@ -211,59 +165,330 @@ func TestDistributeValidatorRewards_ErrorCases(t *testing.T) {
+			validation: func(t *testing.T, result *tokenomicstypes.ClaimSettlementResult) {
+				// Verify equal validator rewards despite different total delegations
+				transfers := result.GetModToAcctTransfers()
+				validatorRewardMap := make(map[string]int64)
+				for _, transfer := range transfers {
+					if transfer.OpReason == tokenomicstypes.SettlementOpReason_TLM_RELAY_BURN_EQUALS_MINT_VALIDATOR_REWARD_DISTRIBUTION {
+						validatorRewardMap[transfer.RecipientAddress] = transfer.Coin.Amount.Int64()
+					}
+				}
+				// Each validator should get 15,000 (300k self-bonded out of 1.8M total)
+				for _, reward := range validatorRewardMap {
+					require.Equal(t, int64(15_000), reward)
+				}
+			},
 		},
 		{
-			OperatorAddress: sample.ValOperatorAddressBech32(),
-			Tokens:          math.NewInt(334), // ~33.4% of total
-			Status:          stakingtypes.Bonded,
+			name: "success: backward compatibility when no delegations found (falls back to validator-only)",
+			validators: []stakingtypes.Validator{
+				createValidator(sample.ValOperatorAddressBech32(), 500_000),
+				createValidator(sample.ValOperatorAddressBech32(), 300_000),
+				createValidator(sample.ValOperatorAddressBech32(), 200_000),
+			},
+			rewardAmount:          math.NewInt(10_000),
+			expectedTransferCount: 3, // validators only
+			opReason:              tokenomicstypes.SettlementOpReason_TLM_GLOBAL_MINT_VALIDATOR_REWARD_DISTRIBUTION,
+			validation: func(t *testing.T, result *tokenomicstypes.ClaimSettlementResult) {
+				transfers := result.GetModToAcctTransfers()
+				// All transfers should be validator rewards
+				for _, transfer := range transfers {
+					require.Equal(t, tokenomicstypes.SettlementOpReason_TLM_GLOBAL_MINT_VALIDATOR_REWARD_DISTRIBUTION, transfer.OpReason)
+				}
+			},
+		},
+		{
+			name: "success: precision handling with Largest Remainder Method for fractional distributions",
+			validators: []stakingtypes.Validator{
+				createValidator(sample.ValOperatorAddressBech32(), 333_333),
+				createValidator(sample.ValOperatorAddressBech32(), 333_333),
+				createValidator(sample.ValOperatorAddressBech32(), 333_334),
+			},
+			rewardAmount:          math.NewInt(100),
+			expectedTransferCount: 6, // 3 validators + 3 delegators
+			opReason:              tokenomicstypes.SettlementOpReason_TLM_GLOBAL_MINT_VALIDATOR_REWARD_DISTRIBUTION,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStakingKeeper := mocks.NewMockStakingKeeper(ctrl)
+
+			// Create delegations inline based on test case
+			delegations := make(map[string][]stakingtypes.Delegation)
+			for i, validator := range tt.validators {
+				valAddr, _ := cosmostypes.ValAddressFromBech32(validator.OperatorAddress)
+
+				switch tt.name {
+				case "success: mixed delegation amounts with equal self-bonded stakes":
```

Comment:
This pattern on matching with test names is weird. What's the rationale/motivation?

Why not prepare the delegations in the test config like we've always done?

Reply:
:thinking: Not sure what happened here. :sweat_smile: